Ensure HTTP/2 connections are gracefully closed on async cancellation (#757)
encode/httpcore: CHANGELOG.md

## Unreleased

> - Fix the unstarted http/2 connection cancellations. (#757)

Reviewer: Perhaps

```suggestion
- Ensure HTTP/2 connections are closed on cancellation. (#757)
```

I think the title of this pull request is a little misleading as well :)

Reviewer 2: Thanks! I changed it to be more descriptive.
